### ISSUE TITLE
feature: donate change output

### ISSUE THREAD

**Issue by cygnet3** [cygnet3/danawallet#96]
Since our wallet is very donation-focused, a good feature would be to suggest to users to donate the change output, rather than return it.

This has a couple of nice features:

- Stimulates donations (which is the goal of Dana)
- Excellent compatibility with the 'contacts' feature. I imagine a user having a list of donation candidates in their contact list. Then we suggest a random candidate from their donation contacts.
- Improves on-chain privacy by breaking the '1-output-1-change' heuristic

**Comment by yashrajd** [comment]
News (for me) but apparently  "chain avoidance" is a [guideline and feature](https://docs.wasabiwallet.io/why-wasabi/Coins.html#change-avoidance-suggestion) Wasabi wallet already has...

**Comment by Sosthene00** [comment]
Yes I think that samourai also had some tricks to avoid change when possible but I might be misremembering. Anyway that all points out the change output being something we want to either avoid or make a donation to mislead chain analysis.

**Comment by yashrajd** [comment]
It'd figure in coin selection which might make it tricky, just a guess. A naive donate change feature should be easy though

**Comment by yashrajd** [comment]
> * Excellent compatibility with the 'contacts' feature. I imagine a user having a list of donation candidates in their contact list. Then we suggest a random candidate from their donation contacts.

Hmm this could also work. But it would mean the output still exists, in which case there is no fee savings. I think the original idea was to incentivise donation by reducing fee through eliminating the change output itself.
